src/de/grobox/liberario/fragments/SettingsFragment.java

```diff
@@ -95,13 +95,8 @@ public boolean onPreferenceClick(Preference preference) {
 		quickhome.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
 			public boolean onPreferenceClick(Preference preference) {
 				// create launcher shortcut
-				Intent shortcutIntent = new Intent(DirectionsFragment.TAG, Uri.EMPTY, getContext(), MainActivity.class);
-				shortcutIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-				shortcutIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-				shortcutIntent.putExtra("special", DirectionsFragment.TASK_BRING_ME_HOME);
-
 				Intent addIntent = new Intent();
-				addIntent.putExtra(Intent.EXTRA_SHORTCUT_INTENT, shortcutIntent);
+				addIntent.putExtra(Intent.EXTRA_SHORTCUT_INTENT, MainActivity.getShortcutIntent(getContext()));
```

Comment:
Thanks for factoring out this code into its own method!

Just a bit strange that the SettingsFragment needs to call a method from the MainActivity. Maybe this should be move to TransportrUtils instead?